app/models/conversation.rb

```diff
@@ -158,10 +159,7 @@ def self_assign?(assignee_id)
   end
 
   def set_display_id
-    self.display_id = loop do
-      next_display_id = account.conversations.maximum('display_id').to_i + 1
-      break next_display_id unless account.conversations.exists?(display_id: next_display_id)
-    end
+    self.display_id = account.conversations.find(id).display_id
```

Comment:
you could do `self.reload` i believe.  
that should update the object with the values from DB 

https://apidock.com/rails/ActiveRecord/Persistence/reload